src/main/java/com/rackspace/ceres/app/services/MetadataService.java

```diff
@@ -123,31 +113,32 @@ public MetadataService(ReactiveCqlTemplate cqlTemplate,
   public Publisher<?> storeMetadata(String tenant, String seriesSetHash,
                                     String metricName, Map<String, String> tags) {
     final CompletableFuture<Boolean> result = seriesSetExistenceCache.get(
-        new SeriesSetCacheKey(tenant, seriesSetHash),
-        (key, executor) ->
-            redisTemplate.opsForValue()
-                .setIfAbsent(
-                    PREFIX_SERIES_SET_HASHES + DELIM + key.getTenant() + DELIM + key
-                        .getSeriesSetHash(),
-                    ""
-                )
-                .doOnNext(inserted -> {
-                  if (inserted) {
-                    redisMiss.increment();
-                  } else {
-                    redisHit.increment();
-                  }
-                })
-                // already cached in redis?
-                .flatMap(inserted -> !inserted ? Mono.just(true) :
-                    // not cached, so store the metadata to be sure
-                    storeMetadataInCassandra(tenant, seriesSetHash, metricName,
-                        tags
-                    )
-                        .map(it -> true))
-                .toFuture()
+            new SeriesSetCacheKey(tenant, seriesSetHash),
+            (key, executor) ->
+                    cassandraTemplate.exists(
```

Comment:
What if we remove this check from here and simply put it in database if metadata is note present in in memory cache?

Reply:
Either way is fine I think. This is a very good improvement.